### ISSUE TITLE
erratic behavior of readTopology with internal node names

### ISSUE THREAD

**Issue by crsl4** [crsl4/PhyloNetworks.jl#85]
[tree.txt](https://github.com/crsl4/PhyloNetworks.jl/files/2363523/tree.txt)
I recently used [ClonalFrameML](https://github.com/xavierdidelot/ClonalFrameML), which produces a tree in parenthetical formal, with internal node names.
I tried to read this tree with PhyloNetworks, with erratic behavior.

If you try to read the tree as is:
```
julia> net=readTopology("tree.txt")
ERROR: Tree ended while reading in subtree beginning with left parenthesis number 125.
Stacktrace:
 [1] readTopology(::IOStream, ::Bool) at /Users/Clauberry/.julia/v0.6/PhyloNetworks/src/readwrite.jl:505
 [2] readTopology(::String, ::Bool) at /Users/Clauberry/.julia/v0.6/PhyloNetworks/src/readwrite.jl:467
 [3] readTopology(::String) at /Users/Clauberry/.julia/v0.6/PhyloNetworks/src/readwrite.jl:483
```
If you open the text file, and add an end-of-line after ; you can read the tree, but the last internal node (NODE_251) is treated as a leaf:
```
julia> net=readTopology("tree.txt")
PhyloNetworks.HybridNetwork, Rooted Network
251 edges
252 nodes: 127 tips, 0 hybrid nodes, 125 internal tree nodes.
tip labels: tx002, tx001, tx049, out, ...
((tx002:0.0,(tx001:0.0,(tx049:0.0,((((((out:0.597,((tx113:0.0,tx170:0.0):0.004,((tx178:0.002,(tx241:0.0,(tx245:0.0,tx243:0.0):0.0):0.002):0.001,((tx199:0.002,(tx205:0.0,tx148:0.0):0.002):0.001,(tx242:0.0,((((tx185:0.0,((tx237:0.0,tx259:0.0):0.0,tx266:0.0):0.0):0.0,(tx230:0.001,tx239:0.0):0.0):0.0,tx225:0.0):0.0,(tx215:0.0,tx217:0.0):0.0):0.0):0.003):0.001):0.0):0.0):0.0,((tx111:0.002,(tx022:0.0,(tx103:0.0,(tx109:0.0,(tx236:0.0,(tx168:0.0,tx169:0.0):0.0):0.0):0.0):0.0):0.002):0.003,(tx253:0.003,((tx207:0.0,((tx222:0.0,tx204:0.0):0.0,(tx104:0.0,(((tx247:0.0,(tx189:0.0,((tx191:0.0,tx177:0.0):0.0,tx264:0.0):0.0):0.0):0.0,tx174:0.0):0.0,((tx180:0.0,((tx383:0.0,tx029:0.0):0.0,tx071:0.0):0.0):0.0,((tx252:0.0,tx162:0.0):0.0,(tx246:0.0,tx112:0.0):0.0):0.0):0.0):0.0):0.0):0.0):0.0,((tx216:0.0,((tx210:0.0,tx223:0.0):0.0,(tx212:0.0,(tx213:0.0,tx196:0.0):0.0):0.0):0.0):0.0,tx219:0.0):0.0):0.003):0.001):0.0):0.002,((tx231:0.001,((tx235:0.0,((tx161:0.0,tx167:0.0):0.0,(tx271:0.0,tx157:0.0):0.0):0.0):0.0,tx232:0.001):0.001):0.002,tx203:0.003):0.001):0.001,((tx265:0.002,((tx220:0.0,tx224:0.0):0.0,tx105:0.0):0.002):0.0,((tx255:0.002,(((tx063:0.0,tx064:0.0):0.0,tx054:0.0):0.001,(((((tx108:0.0,tx172:0.0):0.0,tx254:0.0):0.0,(tx145:0.0,tx114:0.0):0.0):0.0,(tx211:0.0,(tx021:0.0,(tx119:0.0,tx073:0.0):0.0):0.0):0.0):0.0,((tx214:0.0,(tx152:0.0,((tx107:0.0,tx106:0.0):0.0,tx027:0.0):0.0):0.0):0.0,((tx275:0.0,(tx198:0.0,(tx201:0.0,tx202:0.0):0.0):0.0):0.0,(tx100:0.0,((tx218:0.0,tx209:0.0):0.0,(((tx039:0.0,tx249:0.0):0.0,(tx036:0.0,tx272:0.0):0.0):0.0,tx035:0.0):0.0):0.0):0.0):0.0):0.0):0.0):0.001):0.0,(((tx182:0.001,tx256:0.002):0.0,(((tx260:0.0,(tx102:0.0,tx187:0.0):0.0):0.0,tx227:0.0):0.002,(tx175:0.0,tx386:0.0):0.002):0.0):0.0,(((((tx072:0.0,tx226:0.0):0.0,tx194:0.0):0.0,(tx248:0.0,tx192:0.0):0.0):0.001,tx262:0.001):0.001,((tx164:0.0,tx183:0.0):0.0,tx233:0.001):0.001):0.0):0.0):0.0):0.0):0.002,tx070:0.001):0.0,((tx240:0.0,((tx003:0.0,(tx051:0.0,(tx074:0.0,tx079:0.0):0.0):0.0):0.0,tx149:0.0):0.0):0.0,(tx110:0.0,(tx188:0.0,tx387:0.0):0.0):0.0):0.0):0.0):0.0):0.0):0.0,tx268:0.0,NODE_251);
```

**Comment by cecileane** [comment]
similar example, but smaller to make it easier to reproduce both kinds of errors.

First: root name interpreted as a node (leaf) when the newick format ends with a newline
```julia
julia> readTopology("(tx1:0.1,tx2:0.2)rootname;\n")
PhyloNetworks.HybridNetwork, Rooted Network
3 edges
4 nodes: 3 tips, 0 hybrid nodes, 1 internal tree nodes.
tip labels: tx1, tx2, rootname
(tx1:0.1,tx2:0.2,rootname);
```

Second: root name causing an error in the absence of a final newline.
```julia
julia> readTopology("(tx1:0.1,tx2:0.2)rootname;")
ERROR: Tree ended while reading in subtree beginning with left parenthesis number 1.
Stacktrace:
 [1] readTopology(::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at /Users/ane/.julia/v0.6/PhyloNetworks/src/readwrite.jl:505
 [2] readTopology(::String, ::Bool) at /Users/ane/.julia/v0.6/PhyloNetworks/src/readwrite.jl:467
 [3] readTopology(::String) at /Users/ane/.julia/v0.6/PhyloNetworks/src/readwrite.jl:483
```

**Comment by cecileane** [comment]
fixed with 143172347f6c53886d5d78dff35b735b4cfe59d6

```julia
julia> using PhyloNetworks

julia> net = readTopology("(tx1:0.1,tx2:0.2)rootname;\n");

julia> writeTopology(net)
"(tx1:0.1,tx2:0.2)rootname;"

julia> writeTopology(net, internallabel=false)
"(tx1:0.1,tx2:0.2);"

julia> net = readTopology("(tx1:0.1,tx2:0.2)rootname;");

julia> writeTopology(net)
"(tx1:0.1,tx2:0.2)rootname;"

julia> writeTopology(net, internallabel=false)
"(tx1:0.1,tx2:0.2);"
```
